submissions/AnnaStoyano/oop-exercise/index.js

```diff
@@ -0,0 +1,122 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammal {
+   constructor(species, name, gender, legs, friends, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.friends = friends;
+      this.saying = saying;
+   }
+
+   tellAbout(properties) {
+      let propertiesForDispaly = properties || ['species', 'name', 'gender', 'legs', 'friends', 'saying'];
+      return propertiesForDispaly.map(property => `<b>${property}:</b> ${this[property]}`).join('\t');
+   }
+};
+
+class Dog extends Mammal {
+   constructor(name, gender, friends, saying) {
+      super('dog', name, gender, 4, friends, saying || 'guf');
+   }
+
+   tellAbout() {
+      let properties = ['name', 'species', 'friends']; /* properties for display */
+      return super.tellAbout(properties);
+   }
+};
+
+class Cat extends Mammal {
+   constructor(name, gender, friends, saying) {
+      super('cat', name, gender, 4, friends, saying || 'meow');
+   }
+
+   tellAbout() {
+      let properties = ['species', 'name', 'gender', 'legs', 'friends', 'saying']; /* properties for display */
+      return super.tellAbout(properties);
```

Comment:
Below are comments from the past reviews that this implementation doesn't take into account.

> Each class should handle only properties know to it.

> Make the method to work with properties known to this class.
> In every child class overload the method and make it call super.tellAbout() and supplement the result with the values from properties know to that child class.

In current implementation a child class refers to the properties that do not belong to a class.
Current implementation reads "Specify all properties, from current class and all its parent classes and call super class method to manage them" rather than "call super.tellAbout() and supplement the result with the values from properties know to that child class".